### PR TITLE
Proper error message when failing to import 3rd part module

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

**Comment by pbrunet** [comment]
I push a new version. Your fix didn't work as the import can be done by Python but not by Pythran in some case. The test worked as an other PythranSyntaxError was raise later on incorret aliasing or unknown attribute.

**Comment by serge-sans-paille** [comment]
ok, thanks!

**Comment by serge-sans-paille** [comment]
@pbrunet you messed it up, I'm leaving this to you!

**Comment by serge-sans-paille** [comment]
@pbrunet should I have a look on this?

File: pythran/transformations/handle_import.py [hunk 1]
```diff
@@ -355,6 +360,16 @@ def visit_Module(self, module):
 
         # Patch module body: prepend all imported function and import nodes
         imported = self.registry.generate_ImportList()
+        external_modules_name = [f for f in self.module.imported_modules
+                                 if not is_builtin_module_name(f)]
+        if not imported and external_modules_name:
+            # We can't raise an exception as it may just be an unused import.
+            # FIXME : To improve this, wa have to handle aliasing so that we
```
Comment (serge-sans-paille) [inline pythran/transformations/handle_import.py]: wa -> we

File: pythran/transformations/handle_import.py [hunk 2]
```diff
@@ -355,6 +360,16 @@ def visit_Module(self, module):
...
+            # can know sys.stdout is not a attribute of class but a function
```
Comment (serge-sans-paille) [inline pythran/transformations/handle_import.py]: a attribute -> an attribute

File: pythran/transformations/handle_import.py [hunk 3]
```diff
@@ -355,6 +360,16 @@ def visit_Module(self, module):
...
+            # call from a module even when the function is not defined (.so
+            # file for example)
+            print ("Module : '" + "', '".join(external_modules_name) + "' is "
+                   "imported but not use or can't be imported by Pythran.")
```
Comment (serge-sans-paille) [inline pythran/transformations/handle_import.py]: but not use -> but not used

File: pythran/transformations/handle_import.py [hunk 4]
```diff
@@ -355,6 +360,16 @@ def visit_Module(self, module):
...
         imported = self.registry.generate_ImportList()
+        external_modules_name = [f for f in self.module.imported_modules
+                                 if not is_builtin_module_name(f)]
+        if not imported and external_modules_name:
+            # We can't raise an exception as it may just be an unused import.
+            # FIXME : To improve this, wa have to handle aliasing so that we
+            # can know sys.stdout is not a attribute of class but a function
+            # call from a module even when the function is not defined (.so
+            # file for example)
+            print ("Module : '" + "', '".join(external_modules_name) + "' is "
```
Comment (serge-sans-paille) [inline pythran/transformations/handle_import.py]: this is a warning, so use log.warn ?

**Comment by serge-sans-paille** [comment]
you can amend my commit instead of creating a new one, so that there will be just one commit. Apart from the typo, I'm ok with this commit.

File: pythran/tests/test_import_all.py [hunk 5]
```diff
@@ -17,3 +19,19 @@ def test_import_all_twice(self):
...
+
+        Check is done for module as .py file.
+        """
+        code = """
+            import collections
+            def unsupported_module():
+                return collections.Counter()"""
+
+        with self.assertRaises(pythran.syntax.PythranSyntaxError) as ex:
+            pythran.compile_pythrancode("flamby", dedent(code))
```
Comment (serge-sans-paille) [inline pythran/tests/test_import_all.py]: et oui les flamby c'est comme ça tu tires la languette tu sais pas ce qui va se passer ^^!

**Pull request merged**